internal/internal.go

```diff
@@ -608,178 +569,157 @@ const getUserIDSQL = `
-		return
+		return idErr
 	}
 
 	outputMap, err := i.updateTimeLimit(user, id)
 	if err != nil {
-		common.Error(writer, err.Error(), http.StatusInternalServerError)
 		log.Error(err)
-		return
+		return err
 	}
 
-	var outputJSON []byte
-	outputJSON, err = json.Marshal(outputMap)
-	if err != nil {
-		wrappedErr := errors.Wrapf(err, "error marshalling the JSON for the new time limit for analysis %s", id)
-		common.Error(writer, wrappedErr.Error(), http.StatusInternalServerError)
-		log.Error(wrappedErr)
-		return
-	}
+	return c.JSON(http.StatusOK, outputMap)
 
-	fmt.Fprint(writer, string(outputJSON))
 }
 
 // VICEAdminTimeLimitUpdate is basically the same as VICETimeLimitUpdate
 // except that it doesn't require user information in the request.
-func (i *Internal) VICEAdminTimeLimitUpdate(writer http.ResponseWriter, request *http.Request) {
+func (i *Internal) VICEAdminTimeLimitUpdate(c echo.Context) error {
 	var (
-		err   error
-		found bool
-		id    string
-		user  string
+		err  error
+		id   string
+		user string
 	)
 	// id is required
-	if id, found = mux.Vars(request)["analysis-id"]; !found {
-		common.Error(writer, errors.New("id parameter is empty").Error(), http.StatusBadRequest)
-		return
+	id = c.Param("analysis-id")
+	if id == "" {
+		return &echo.HTTPError{
+			Code:     http.StatusBadRequest,
+			Internal: errors.New("id parameter is empty"),
+		}
 	}
 
 	apps := apps.NewApps(i.db)
 
 	user, _, err = apps.GetUserByAnalysisID(id)
 	if err != nil {
-		common.Error(writer, err.Error(), http.StatusInternalServerError)
-		return
+		return err
 	}
 
 	outputMap, err := i.updateTimeLimit(user, id)
 	if err != nil {
-		common.Error(writer, err.Error(), http.StatusInternalServerError)
-		return
-	}
-
-	var outputJSON []byte
-	outputJSON, err = json.Marshal(outputMap)
-	if err != nil {
-		common.Error(writer, errors.Wrapf(err, "error marshalling the JSON for the new time limit for analysis %s", id).Error(), http.StatusInternalServerError)
-		return
+		return err
 	}
 
-	fmt.Fprint(writer, string(outputJSON))
+	return c.JSON(http.StatusOK, outputMap)
 }
 
 // VICEGetTimeLimit implements the handler for getting the current time limit from the database.
-func (i *Internal) VICEGetTimeLimit(writer http.ResponseWriter, request *http.Request) {
+func (i *Internal) VICEGetTimeLimit(c echo.Context) error {
 	log.Info("get time limit called")
 
 	var (
 		err        error
 		analysisID string
-		users      []string
 		user       string
 		userID     string
-		found      bool
 	)
 
 	// user is required
-	if users, found = request.URL.Query()["user"]; !found {
-		common.Error(writer, "user is not set", http.StatusForbidden)
-		return
+	user = c.QueryParam("user")
+	if user == "" {
+		return &echo.HTTPError{
+			Code:     http.StatusForbidden,
+			Internal: errors.New("user is not set"),
+		}
 	}
-	user = users[0]
 
 	if !strings.HasSuffix(user, "@iplantcollaborative.org") {
 		user = fmt.Sprintf("%s@iplantcollaborative.org", user)
 	}
```

Comment:
I'd like to see a configuration setting for this. Not all DE deployments use `iplantcollaborative.org` as the username domain.

Reply:
I made this a command-line flag for now, adding it to job-services.yaml shouldn't be terrible if needed in the future.